internal/worker/save_data.go

```diff
@@ -17,8 +17,11 @@ type ResultStores struct {
 	DynamicAnalysis *resultstore.ResultStore
 	StaticAnalysis  *resultstore.ResultStore
 	FileWrites      *resultstore.ResultStore
+	AnalyzedPackage *resultstore.ResultStore
 }
 
+var analyzedPackageSaved = false
```

Comment:
This state should not be global; let's move it into the `ResultStores` struct.

The flag will need to be reset to `false` in `handleMessage` in cmd/worker/main.go